src/lib.rs

```diff
@@ -119,64 +114,29 @@ impl Bloom {
     ///
     fn contains<T: Hash>(&self, item: &T) -> bool {
         self.index_iterator(item)
-            .all(|index| *self.field.get(index).unwrap())
-    }
-
-    /// Test the fill ratio of the Bloom
-    ///
-    /// # Arguments
-    ///
-    /// * `lower_bound` - The minimum fill ratio
-    ///
-    /// # Example
-    ///
-    ///
-    /// let bloom = Bloom::new(0.05, 10);
-    /// let item = 0;
-    ///
-    /// bloom.insert(&item);
-    /// assert!(!bloom.fill_ratio_gte(0.01));
-    ///
-    fn fill_ratio_gte(&self, lower_bound: f64) -> bool {
-        let len = (self.slice_len * self.num_slices) as f64;
-        (self.field.count_ones() as f64 / len) >= lower_bound
+            .all(|(byte, mask)| self.buffer[byte] & mask != 0)
     }
-
-    /// Clear the bloom filter.
-    fn clear(&mut self) {
-        self.field.iter_mut().for_each(|mut b| *b = false);
-    }
-}
-
-/// Convenience function to hash a u64
-///
-/// # Arguments
-///
-/// * `i` - The u64 to hash
-fn hash_u64(i: u64) -> u64 {
-    seahash::hash(&i.to_be_bytes())
 }
 
-/// Stretch a u64 into four u64s
-///
-/// # Arguments
-///
-/// * `base_seed` - The seed to stretch
-#[inline]
-fn generate_seed(base_seed: u64) -> (u64, u64, u64, u64) {
-    let h1 = hash_u64(base_seed);
-    let h2 = hash_u64(h1);
-    let h3 = hash_u64(h2);
-    let mut hasher = SeaHasher::with_seeds(base_seed, h1, h2, h3);
-    hasher.write_u64(0);
-    let a = hasher.finish();
-    hasher.write_u64(a);
-    let b = hasher.finish();
-    hasher.write_u64(b);
-    let c = hasher.finish();
-    hasher.write_u64(c);
-    let d = hasher.finish();
-    (a, b, c, d)
+/// Returns 2 hashes for the given item
+fn double_hashing_hashes<T: Hash>(item: T) -> (usize, usize) {
+    let mut hs = ahash::AHasher::new_with_keys(
+        0xe7b0c93ca8525013011d02b854ae8182,
+        0x7bcc5cf9c39cec76fa336285d102d083,
+    );
+    item.hash(&mut hs);
+    let h1 = hs.finish();
+
+    hs = ahash::AHasher::new_with_keys(
+        0x16f11fe89b0d677cb480a793d8e6c86c,
+        0x6fe2e5aaf078ebc914f994a4c5259381,
+    );
+    item.hash(&mut hs);
+
+    // h2 hash shouldn't be 0 for double hashing
+    let h2 = hs.finish().max(1);
+
+    (h1 as _, h2 as _)
```

Comment:
It's kinda ugly but a common convention in my code bases is to not use implicit conversions like this. It's a-ok here, so read-and-resolve